src/test/scala/chiselTests/InlineSpec.scala

```diff
@@ -0,0 +1,124 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import chisel3.util.{InlineInstance, InlineHelpers}
+import firrtl.FirrtlExecutionSuccess
+import firrtl.passes.InlineAnnotation
+import firrtl.{ir => fir}
+import firrtl.WDefInstance
+import firrtl.Mappers._
+import org.scalatest.{FreeSpec, Matchers}
+
+class InlineSpec extends FreeSpec with ChiselRunners with Matchers {
+
+  trait Internals { this: Module =>
+    val io = IO(new Bundle{ val a = Input(Bool()) })
+  }
+
+  class Foo extends Module with Internals with InlineHelpers with InlineInstance
+  class Bar extends Module with Internals with InlineHelpers
+
+  def collectInstances(c: fir.Circuit): Array[WDefInstance] = {
+    val insts = scala.collection.mutable.ArrayBuffer[WDefInstance]()
+    def onModule(m: fir.DefModule): fir.DefModule = {
+      def onStmt(s: fir.Statement): fir.Statement = s map onStmt match {
+        case i: WDefInstance => insts += i
+          s
+        case _ =>
+          s
+      }
+      m map onStmt
+      m
+    }
+    c map onModule
+    insts.toArray.map{ case WDefInstance(_, n, m, _) => WDefInstance(fir.NoInfo, n, m, fir.UnknownType) }
```

Comment:
Alternatively:
```scala
insts.toArray.map(inst => inst.copy(info = fir.NoInfo, tpe = fir.UnknownType))
```

Reply:
Much cleaner, thanks! I can even shave off the `inst => inst.copy` --> `_.copy` ...